### PR TITLE
Feat/6737 data workloads testing

### PR DESCRIPTION
**Pull request by adurkan-amido**
#### 📲 What

Docs updates: addition of testing content for data workloads, tidy up of pysparkle data quality.

#### 🤔 Why
		
Docs updates.
		
#### 🛠 How
		
Docs updates.

#### 👀 Evidence
		
Docs updates, testing in local server.
		 
#### 🕵️ How to test

Run in local server, npm run lint.

#### ✅ Acceptance criteria Checklist

- [x] Code peer reviewed?
- [x] Documentation has been updated to reflect the changes?
- [x] Passing all automated tests, including a successful deployment?
- [x] Passing any exploratory testing?
- [x] Rebased/merged with latest changes from development and re-tested?
- [x] Meeting the Coding Standards?

### CODE REVIEW

File: docs/workloads/azure/data/etl_pipelines/pysparkle/pysparkle_quickstart.md
Comment (adurkan-amido) [inline docs/workloads/azure/data/etl_pipelines/pysparkle/pysparkle_quickstart.md:1]: moved to `pysparkle.md`

File: docs/workloads/azure/data/etl_pipelines/pysparkle/pysparkle_data_quality.md
Comment (adurkan-amido) [inline docs/workloads/azure/data/etl_pipelines/pysparkle/pysparkle_data_quality.md:1]: content moved to `data_quality_azure.md`

File: docs/workloads/azure/data/etl_pipelines/data_quality_azure.md [hunk 1]
```diff
@@ -35,3 +34,89 @@ The design of the Data Quality processing is outlined in the following diagram.
...
+```json
+{
+    "gx_directory_path": "/dbfs/great_expectations/",
+    "dataset_name": "movies_dataset",
+    "dq_output_path": "abfss://staging@{ADLS_ACCOUNT}.dfs.core.windows.net/Ingest_AzureSql_Example/",
+    "datasource_config": [
+        {
+            "datasource_name": "movies_metadata",
+            "datasource_type": "table",
+            "data_location": "staging.movies_metadata",
```
Comment (Danphillipz) [inline docs/workloads/azure/data/etl_pipelines/data_quality_azure.md:90]: Is this a valid example? Would you not need to specify the full path, i.e `abfss://raw@{ADLS_ACCOUNT}.dfs.core.windows.net/pipeline_name/schema.table/v#/*/*/*/*`

**Review by sam-deeble — DISMISSED** [review]

File: docs/workloads/azure/data/etl_pipelines/data_quality_azure.md [hunk 2]
```diff
@@ -35,3 +34,89 @@ The design of the Data Quality processing is outlined in the following diagram.
...
+          * `"abfss://raw@accountname.dfs.core.windows.net/myfolder/mysubfolder/*"`,
+          * `"abfss://staging@{ADLS_ACCOUNT}.dfs.core.windows.net/myfolder/mysubfolder/*"`.
+       * **Table Name**: For tables with metadata managed by a data catalog, you should provide
+       the database schema and the table name. For example, `staging.table_name`.
+    4. `expectation_suite_name`: Name of the expectation suite associated with this data source.
+    5. `validation_config`: A list of validation configurations where each configuration contains the following fields:
+        1. `column_name`: Name of the validated column.
+        2. `expectations`: List of expectations where each expectation has the following fields:
+        3. `expectation_type`: Name of the Great Expectations expectation class to use.
+        4. `expectation_kwargs`: The keyword arguments to pass to the expectation class.
```
Comment (Danphillipz) [inline docs/workloads/azure/data/etl_pipelines/data_quality_azure.md:75]:
```suggestion
        2. `expectations`: List of expectations where each expectation has the following fields:
            - `expectation_type`: Name of the Great Expectations [expectation class](https://greatexpectations.io/expectations/) to use.
            - `expectation_kwargs`: The keyword arguments to pass to the expectation class.
```

**Review by Danphillipz — CHANGES_REQUESTED** [review]
Much improved Andy, great work on the docs!

File: docs/workloads/azure/data/etl_pipelines/data_quality_azure.md (hunk 1 above)
Comment (adurkan-amido) [reply docs/workloads/azure/data/etl_pipelines/data_quality_azure.md:90]: I believe this syntax is correct if was using a `table` datasource_type - however we're not currently saving data as tables (@lorraine-houston-ensono could you confirm that?) so agree it'd be better to show an similar to the ones we have with parquet or delta.

File: docs/workloads/azure/data/etl_pipelines/data_quality_azure.md [hunk 3]
```diff
@@ -35,3 +34,89 @@ The design of the Data Quality processing is outlined in the following diagram.
...
+The configuration is defined using Python's Pydantic library for data validation.
+
+Here is the description of the main elements:
+
+1. `gx_directory_path`: Path to the Great Expectations metadata store.
+2. `dataset_name`: Name of the dataset that is being processed.
+3. `datasource_config`: List of datasource configurations where each configuration contains the following fields:
+    1. `datasource_name`: Name of the data asset, e.g., table or file name.
+    2. `datasource_type`: Source system type that Spark can read from, e.g. table, parquet, csv, delta.
+    3. `data_location`: Location of the given data asset. It can either be a path to the data file
```
[the remark this answers is not in the document]
Comment (adurkan-amido) [reply docs/workloads/azure/data/etl_pipelines/data_quality_azure.md:61]: Do you not think the examples below specify this enough? I guess this assumes they know where the data is saved that they want to run the checks on.

**Review by Danphillipz — APPROVED** [review]

**Pull request merged**